docs/notes.md

```diff
@@ -224,3 +224,94 @@ We agreed to use hyphen-sepraten name covention for custom Events i.e.
+parameter "scheduler" indicates the interval on which a given request is called.
+ 
+##### General flow:
+ 
+Query Manager accepts:
+- **ApiHelper** - takes care of API calls, it's instantiated on app start(in the `Injector.tsx` file) before all other Managers/Resolvers which then is passed down to ones that need that helper.
+- **StateHelper** - get call status and eventual data assigned[^1], it's also instantiated on the start, like **ApiHelper**
+- **Scheduler** - as mentioned above sets the interval on which query is asked[^2]
+- **getUnique** - a callback that returns a unique "id" that is being populated to Scheduler to avoid having duplicates of queries in scheduler[^2]
+- **getDependencies** - a callback that returns dependencies on which that call relies (with env or not)
+- **kind** - defines what kind of call it is from a set of predefined calls[^1]
+- **getUrl** - a callback that returns the correct URL tail for a given call[^1]
+- **toUsed** - a callback that is used to process data after retrieved from a call
+- **strategy** - a string enum that is used as one of the conditions to define whether to set query call status to loading or not[^3]
+ 
+[^1]: These parameters can be with or without Env, depending if the call requires one
+[^2]: parameter for the Continous Query manager **only**
+[^3]: parameter for the One Time Query manager **only**
+ 
+Each query Manager has:
+- internal **update** function which takes care of appending data to stateHelper
+- **useOneTime** hook which has two useEffect, one listens for query change and update URL, second listens to URL change to perform fire update() function, also returns data and callback that essentially run update function which is used across the app as retry function
+- matches function that checks given query.kind with one that is initially sent to query manager(with kind parameter on initialization)
+ 
+Each initialization of Query Manager is appended to the main Query Resolver which stores every instance of the queries available in the app. Query Resolver on the other hand is stored in DependecyContext with other Managers or Resolvers.
+ 
+These instances can be used in components as in the example below:
+ 
+```
+const { queryResolver } = useContext(DependencyContext);
+const [statusData, retry] = queryResolver.useOneTime<"GetServerStatus">({
+ kind: "GetServerStatus",
+});
+```
+ 
+All **queryManagers** are stored in QueryResolver class which allows us to browse through the ready-to-use implementation of each query only with kind parameters as seen above.
+After that assignment we get an Array from the hook with RemoteData Object which, in short, has 4 statuses:
+1. Not Asked
+2. Loading
+3. Failed - with the value that stores an error message
+4. Success- with the value that corresponds to data requested in a query
+ 
+That object has to be unfolded to access data through RemoteData.unfold()
```

Comment:
.fold() 
Sorry if my previous comment confused you, it wasn't meant to make you change the name of the method, since .fold was correct. But it was more like an observation on the irony of the naming and description.

Reply:
got confused as I didn't notice and remember that I changed that finally :D